MaxText/standalone_dataloader.py

```diff
@@ -38,22 +36,20 @@ def data_load_loop(config, state=None):
   """Main data loader loop.
     Loads batches of data for each training step.
   """
-  _, _, _, _, _, _, _, data_iterator, state = setup_train_loop(config)
+  _, _, _, _, _, _, _, data_iterator, _, state = setup_train_loop(config)
 
-  devices_array = max_utils.create_device_mesh(config)
-  mesh = Mesh(devices_array, config.mesh_axes)
   example_batch = None
 
   start = datetime.datetime.now()
   start_step = get_first_step(state)
-  example_batch = load_next_batch(data_iterator, example_batch, config, mesh)
```

Comment:
please talk to @aireenmei -- not sure why you all are taking and removing mesh :-)

Reply:
Thank you @aireenmei & @rwitten.

`load_next_batch` function is folded in [`MultiHostDataLoadIterator`](https://github.com/google/maxtext/blob/de24bde15c264704c99a12c433a14d8aa75cebed/MaxText/multihost_dataloading.py#L96) which accepts both `tf.data.dataset` and `grain.DataLoader`. `mesh` is not needed since it is part of the class. 

The main goal is to ensure repeatable evaluation loop. We may trigger eval run multiple times and need a way to [reset evaluation iterator](https://github.com/google/maxtext/blob/de24bde15c264704c99a12c433a14d8aa75cebed/MaxText/tests/tfds_data_processing_test.py#L128)

Feel free to let me know what you think.

Reply:
Thanks for the explanation. LGTM